src/ImageSharp/Formats/Tiff/TiffFrameMetadata.cs

```diff
@@ -0,0 +1,292 @@
+                    return TiffConstants.RowsPerStripInfinity;
+                }
+
+                return rowsPerStrip.Value;
+            }
+        }
+
+        /// <summary>
+        /// Gets for each strip, the number of bytes in the strip after compression.
+        /// </summary>
+        public Number[] StripByteCounts
+        {
+            get
+            {
+                IExifValue<Number[]> stripByteCounts = this.ExifProfile.GetValue(ExifTag.StripByteCounts);
+                if (stripByteCounts == null)
+                {
+                    TiffThrowHelper.ThrowImageFormatException("StripByteCounts are missing");
+                }
+
+                return stripByteCounts.Value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the resolution of the image in x- direction.
+        /// </summary>
+        /// <value>The density of the image in x- direction.</value>
+        public double? HorizontalResolution => this.ExifProfile.GetValue(ExifTag.XResolution)?.Value.ToDouble();
+
+        /// <summary>
+        /// Gets the resolution of the image in y- direction.
+        /// </summary>
+        /// <value>The density of the image in y- direction.</value>
+        public double? VerticalResolution => this.ExifProfile.GetValue(ExifTag.YResolution)?.Value.ToDouble();
+
+        /// <summary>
+        /// Gets how the components of each pixel are stored.
+        /// </summary>
+        public TiffPlanarConfiguration PlanarConfiguration => (TiffPlanarConfiguration?)this.ExifProfile.GetValue(ExifTag.PlanarConfiguration)?.Value ?? DefaultPlanarConfiguration;
+
+        /// <summary>
+        /// Gets the unit of measurement for XResolution and YResolution.
+        /// </summary>
+        public PixelResolutionUnit ResolutionUnit => UnitConverter.ExifProfileToResolutionUnit(this.ExifProfile);
+
+        /// <summary>
+        /// Gets a color map for palette color images.
+        /// </summary>
+        public ushort[] ColorMap => this.ExifProfile.GetValue(ExifTag.ColorMap)?.Value;
+
+        /// <summary>
+        /// Gets the description of extra components.
+        /// </summary>
+        public ushort[] ExtraSamples => this.ExifProfile.GetValue(ExifTag.ExtraSamples)?.Value;
+
+        /// <summary>
+        /// Gets a mathematical operator that is applied to the image data before an encoding scheme is applied.
+        /// </summary>
+        public TiffPredictor Predictor => (TiffPredictor?)this.ExifProfile.GetValue(ExifTag.Predictor)?.Value ?? DefaultPredictor;
+
+        /// <summary>
+        /// Gets the specifies how to interpret each data sample in a pixel.
+        /// <see cref="SamplesPerPixel"/>
+        /// </summary>
+        public TiffSampleFormat[] SampleFormat => this.ExifProfile.GetValue(ExifTag.SampleFormat)?.Value?.Select(a => (TiffSampleFormat)a).ToArray();
+
+        /// <summary>
+        /// Clears the pure metadata.
+        /// </summary>
+        public void ClearMetadata()
```

Comment:
This method should not exist

Reply:
I don't know without a deep dig into the source code. What I do know is that there should be absolutely no readonly properties in the class. Metadata should be mutable.

Why are we not simply exposing the ExifProfile?

Reply:
I have changed it so the ExifPofile of an image will be a deep clone of the first frame ExifProfile: https://github.com/SixLabors/ImageSharp/blob/875db673866d44f7ff791cabeeccec5001d92235/src/ImageSharp/Formats/Tiff/TiffDecoderMetadataCreator.cs#L59

Also i have now remove Tiff specific tags from the profile: 875db673

What do you guys think, is that ok like this?

Reply:
So it's a separate instance... Is it required then or should we just expose and allow mutation of each individual frame ExifProfile?

Reply:
I thought it would make it more convenient for the user this way, but the more i think about it the more problems i can imagine. We would have two separate ExifProfiles the user could alter one in the Image metadata and then in the RootFrame another one which we need to sync somehow.

Lets leave the ExifProfile in each image frame which the user can mutate. I will revert and commit the changes soon.